### PR TITLE
Update grpcClient configuration attributes

### PR DESCRIPTION
**Pull request by wtlucy**
A few updates need to be made to the `<grpcTarget/>` configuration:
* rename `grpcTarget` -> `grpcClient`
* rename `sslConfig` -> `sslRef` and add correct type 
* add new props `keepAliveWithoutCalls`, `maxInboundMetadataSize`, `overrideAuthority`, `userAgent`, and `usePlaintext`
* remove `enableKeepAlive` and `authnToken`
* set duration, min, max, default where applicable

for #10089

### CODE REVIEW

**Comment by wtlucy** [comment]
#build

**Comment by skasund** [comment]
L2 message review not required

**Comment by wtlucy** [comment]
#build

**Review by anupag — APPROVED** [review]
looks good other than what was discussed in Slack chat

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#build

File: dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -18,29 +18,42 @@
 clientConfig=GRPC Client Properties
 clientConfig.desc=Configuration properties to be applied to gRPC targets that match the specified URI.
 
-target=Remote target
-target.desc=The remote gRPC target URI, with wildcard support.
+host=Remote host
+host.desc=The remote gRPC service hostname or IP address, with wildcard support.
+
+path=Remote procedure path
+path.desc=The remote gRPC service path, with wildcard support. A path consists of the gRPC service and method names, in the form "ServiceName/MethodName".
```
Comment (Charlotte-Holt) [inline dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties:25]: Suggested update 

>A path consists of the gRPC service and method names, in the form "ServiceName/MethodName".

To

>A path consists of the gRPC service and method names, in the "ServiceName/MethodName" format.

File: dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -18,29 +18,42 @@
...
-enableKeepAlive.desc=Enable gRPC keepalive.
-
 keepAliveTime=Keepalive interval
 keepAliveTime.desc=The time to wait for new messages before sending a new keepalive ping.
 
 keepAliveTimeout=Keepalive timeout
 keepAliveTimeout.desc=End the connection if a keepalive response is not received within this time.
 
+keepAliveWithoutCalls=Keepalive without calls
+keepAliveWithoutCalls.desc=Perform keepalive when there are no outstanding RPCs on the channel.
```
Comment (Charlotte-Holt) [inline dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties:40]: Suggested update

>Perform keepalive when there are no outstanding RPCs on the channel.

To

>Perform keepalive when no outstanding RPCs are on the channel.

File: dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties [hunk 3]
```diff
@@ -18,29 +18,42 @@
...
+
 maxInboundMessageSize=Maximum message size
 maxInboundMessageSize.desc=The maximum inbound message size.
  
-sslConfig=The id of an ssl reference in server.xml
-sslConfig.desc=An id of an ssl reference found in server.xml. The ssl configuration that is specified by this reference is used in the gRPC client configuration.
+sslRef=SSL Reference
```
Comment (Charlotte-Holt) [inline dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties:45]: Suggested update

>SSL Reference

To

>SSL reference

File: dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties [hunk 4]
```diff
@@ -18,29 +18,42 @@
...
+sslRef.desc=Specifies an ID of the SSL configuration that is used to connect to the gRPC service.
+sslRef$Ref=SSL reference
+
+maxInboundMetadataSize=Maximum metadata size
+maxInboundMetadataSize.desc=The maximum allowed inbound metadata size.
+
+overrideAuthority=Override authority
+overrideAuthority.desc=A custom authority which will override the default authority.
```
Comment (Charlotte-Holt) [inline dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties:53]: Suggested update

>A custom authority which will override the default authority.

To

>A custom authority that overrides the default authority.

File: dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties [hunk 5]
```diff
@@ -18,29 +18,42 @@
...
+
+userAgent=Custom user agent
+userAgent.desc=A custom User-Agent which will prepend the default User-Agent.
```
Comment (Charlotte-Holt) [inline dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties:56]: Suggested update

>A custom User-Agent which will prepend the default User-Agent.

To

>A custom User-Agent that adds a prefix to the default User-Agent.

File: dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties [hunk 6]
```diff
@@ -18,29 +18,42 @@
...
+
+usePlaintext=Use a plaintext connection
```
Comment (Charlotte-Holt) [inline dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties:58]: Suggested update

>Use a plaintext connection

To

>Use a plain text connection

File: dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties [hunk 7]
```diff
@@ -18,29 +18,42 @@
...
+usePlaintext.desc=Use a plaintext connection for the outbound gRPC channel.
```
Comment (Charlotte-Holt) [inline dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties:59]: Suggested update

>Use a plaintext connection for the outbound gRPC channel.

To

>Use a plain text connection for the outbound gRPC channel.

**Review by Charlotte-Holt — COMMENTED** [review]
@wtlucy I added a few ID review comments. Let me know if you have any questions! Thx!

File: dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties (hunk 3 above)
Comment (wtlucy) [reply dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties:45]: I agree that making reference lowercase here makes the most sense - but the build process includes a schema validator tool that fails if I make this change.

Comment (wtlucy) [reply dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties:45]: `junit.framework.AssertionFailedError: The label "SSL reference" for element ssl of pid io.openliberty.grpc.clientConfig should be in title case.`

**Comment by wtlucy** [comment]
Thanks for reviewing @Charlotte-Holt - I've made the requested changes

File: dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties (hunk 3 above)
Comment (Charlotte-Holt) [reply dev/io.openliberty.grpc.1.0.internal.client/resources/OSGI-INF/l10n/metatype.properties:45]: Ah, okay. Then it's alright.

**Comment by wtlucy** [comment]
#build

**Comment by wtlucy** [comment]
#run-libby-bot

**Pull request merged**